qa/tasks/cephfs/test_client_recovery.py

```diff
@@ -629,3 +630,73 @@ def test_config_session_timeout(self):
         self.assert_session_count(1)
 
         self.mount_a.kill_cleanup()
+
+    def test_reconnect_after_blacklisted(self):
+        """
+        Test reconnect after blacklisted.
+        - writing to a fd that was opened before blacklist should return -EBADF
+        - reading/writing to a file with lost file locks should return -EIO
+        - readonly fd should continue to work
+        """
+
+        if not isinstance(self.mount_a, KernelMount):
+            self.skipTest("Not implemented in FUSE client yet")
+
+        self.mount_a.umount_wait()
+        self.mount_a.mount(mount_opts=['recover_session=clean'])
+        self.mount_a.wait_until_mounted()
+
+        path = os.path.join(self.mount_a.mountpoint, 'testfile_reconnect_after_blacklisted')
+        pyscript = dedent("""
+            import os
+            import sys
+            import fcntl
+            import errno
+            import time
+
+            fd1 = os.open("{path}.1", os.O_RDWR | os.O_CREAT, 0666)
+            fd2 = os.open("{path}.1", os.O_RDONLY)
+            fd3 = os.open("{path}.2", os.O_RDWR | os.O_CREAT, 0666)
+            fd4 = os.open("{path}.2", os.O_RDONLY)
+
+            os.write(fd1, 'content')
+            os.read(fd2, 1);
+
+            os.write(fd3, 'content')
+            os.read(fd4, 1);
+            fcntl.flock(fd4, fcntl.LOCK_SH | fcntl.LOCK_NB)
+
+            sys.stdout.write('\n')
+            sys.stdin.readline()
+
+            time.sleep(10);
```

Comment:
Do we need this sleep? The stdin/stdout I/O should be enough.

Reply:
10s is for client auto reconnect (because auto reconnect is driven by tick).  write syscall before auto reconnect still return error.

Reply:
If the client gets an error from an OSD that it's been blacklisted, shouldn't it try to initiate the reconnect before returning some kind of error to the application?

Reply:
writable file handle become invalid after reconnect. Application will get error eventually.